### PR TITLE
Debug subsystem tidying

### PR DESCRIPTION
**Pull request by drewnoakes**
Some refactoring and tidy ups made while watching the showcase.

The biggest change here is the simplification of the dataflow source in [DebugProfileDebugTargetGenerator](https://github.com/dotnet/project-system/commit/c5c1c533a9a25a5dc0f8dc04bda72d6224597060), which deletes a bunch of code and avoids registering the data source.

Other than that, reduced allocations, improved docs, formatting.

Easiest to review commit-by-commit.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8959)

### CODE REVIEW

**Comment by drewnoakes** [comment]
The build failure seems unrelated to this PR:

> The repository primary signature validity period has expired.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/DebugProfileDebugTargetGenerator.cs [hunk 1]
```diff
@@ -15,101 +16,49 @@ namespace Microsoft.VisualStudio.ProjectSystem.Debug
     [AppliesTo(ProjectCapability.LaunchProfiles)]
     [Export(typeof(IDynamicDebugTargetsGenerator))]
     [ExportMetadata("Name", "DebugProfileProvider")]
-    internal class DebugProfileDebugTargetGenerator : ProjectValueDataSourceBase<IReadOnlyList<IEnumValue>>, IDynamicEnumValuesProvider, IDynamicDebugTargetsGenerator
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/DebugProfileDebugTargetGenerator.cs:18]: This component was changed from `ProjectValueDataSourceBase<>` to `ChainedProjectValueDataSourceBase<>`.

The latter is a simple alternative when you are producing data that's entirely derived from other data sources.

**Review by haileymck — APPROVED** [review]

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/DebugProfileDebugTargetGenerator.cs (hunk 1 above)
Comment (melytc) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/DebugProfileDebugTargetGenerator.cs:18]: What are the data sources this generator is getting the data from? Is one source the `IVersionedLaunchSettingsProvider`?

**Review by melytc — APPROVED** [review]
looks good!

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/DebugProfileDebugTargetGenerator.cs (hunk 1 above)
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/DebugProfileDebugTargetGenerator.cs:18]: `ChainedProjectValueDataSourceBase` has one abstract method you have to implement where you link upstream data source(s) to the target block you are passed. You can see the sources in that method.

**Pull request merged**